tests/filter.json

```diff
@@ -932,6 +932,30 @@
       "selector" : "$[?(@.a || @.b) && @.b]",
       "document" : [{"a": 1}],
       "result": []
+    },
+    {
+      "name": "string literal, single quote in double quotes",
+      "selector" : "$[?@ == \"quoted' literal\"]",
+      "document" : [{"a": 1}],
+      "result": []
+    },
+    {
+      "name": "string literal, double quote in single quotes",
+      "selector" : "$[?@ == 'quoted\" literal']",
+      "document" : [{"a": 1}],
+      "result": []
+    },
+    {
+      "name": "string literal, escaped single quote in single quotes",
+      "selector" : "$[?@ == 'quoted\\' literal']",
+      "document" : [{"a": 1}],
+      "result": []
+    },
+    {
+      "name": "string literal, escaped double quote in double quotes",
+      "selector" : "$[?@ == \"quoted\\\" literal\"]",
+      "document" : [{"a": 1}],
+      "result": []
     }
```

Comment:
How about add tests for named selector too

Reply:
Those already exist.